secure_message/repository/retriever.py

```diff
@@ -46,38 +52,95 @@ def retrieve_message_list(page, limit, user, ru_id=None, survey=None, cc=None, c
 
         try:
             t = db.session.query(SecureMessage.msg_id, func.max(Events.date_time)  # pylint:disable=no-member
-                                 .label('max_date'))\
+                                 .label('max_date')) \
                 .join(Events).join(Status).outerjoin(Actors) \
                 .filter(and_(*conditions)) \
                 .filter(and_(*status_conditions)) \
                 .filter(or_(Events.event == EventsApi.SENT.value, Events.event == EventsApi.DRAFT_SAVED.value)) \
                 .group_by(SecureMessage.msg_id).subquery('t')
 
             if descend:
-                result = SecureMessage.query\
-                    .filter(SecureMessage.msg_id == t.c.msg_id) \
-                    .order_by(t.c.max_date.desc()).paginate(page, limit, False)
-
+                order = t.c.max_date.desc()
             else:
-                result = SecureMessage.query \
-                    .filter(SecureMessage.msg_id == t.c.msg_id) \
-                    .order_by(t.c.max_date.asc()).paginate(page, limit, False)
+                order = t.c.max_date.asc()
+
+            result = SecureMessage.query \
+                .filter(SecureMessage.msg_id == t.c.msg_id) \
+                .order_by(order).paginate(page, limit, False)
 
         except Exception as e:
             logger.error('Error retrieving messages from database', error=e)
             raise InternalServerError(description="Error retrieving messages from database")
 
         return True, result
 
+    @staticmethod
+    def _retrieve_message_list_internal(page, limit, ru_id, survey, cc, ce, label, descend):
+        """returns list of messages from db"""
+        conditions = []
+        status_reject_conditions = []
+        valid_statuses = []
+        actor_conditions = []
+
+        if label is not None:
+            valid_statuses.append(label)
+            if label in [Labels.INBOX.value, Labels.ARCHIVE.value, Labels.UNREAD.value]:
+                actor_conditions.append(Actors.sent_from_internal == False)  # NOQA pylint:disable=singleton-comparison
```

Comment:
This to me sums up why the DB structure isn't correct because it shouldn't be this complicated to determine if the internal user has sent or received the message

Reply:
Thats not what its doing . Its adding predicates specifically only when someone has asked for only messages with specific labels to be returned . The simple sent_from_internal flag tells us if it was sent by an internal person

Reply:
Seems I misunderstood it then